src/Network/Network.Test/ScenarioTests/NatGatewayTests.ps1

```diff
@@ -0,0 +1,293 @@
+    }
+}
+
+<#
+.SYNOPSIS
+Test creating new NatGateway using minimal set of parameters
+#>
+function Test-NatGatewayWithSubnet
+{
+    # Setup
+    $rgname = Get-ResourceGroupName;
+    $rglocation = Get-ProviderLocation ResourceManagement;
+    $rname = Get-ResourceName;
+	$vnetName = Get-ResourceName
+    $subnetName = Get-ResourceName
+    $location = "eastus2";
+	$sku = "Standard"
+
+    try
+    {
+        $resourceGroup = New-AzResourceGroup -Name $rgname -Location $rglocation;
+
+        # Create NatGateway
+        $vNatGateway = New-AzNatGateway -ResourceGroupName $rgname -Name $rname -Location $location -sku $sku;
+        Assert-NotNull $vNatGateway;
+        Assert-True { Check-CmdletReturnType "New-AzNatGateway" $vNatGateway };
+        Assert-AreEqual $rname $vNatGateway.Name;
+
+        # Get NatGateway
+        $vNatGateway = Get-AzNatGateway -ResourceGroupName $rgname -Name $rname;
+        Assert-NotNull $vNatGateway;
+        Assert-True { Check-CmdletReturnType "Get-AzNatGateway" $vNatGateway };
+        Assert-AreEqual $rname $vNatGateway.Name;
+
+        # Get all NatGateways in resource group
+        $listNatGateway = Get-AzNatGateway -ResourceGroupName $rgname;
+        Assert-NotNull ($listNatGateway | Where-Object { $_.ResourceGroupName -eq $rgname -and $_.Name -eq $rname });
+
+        # Get all NatGateways in subscription
+        $listNatGateway = Get-AzNatGateway;
+        Assert-NotNull ($listNatGateway | Where-Object { $_.ResourceGroupName -eq $rgname -and $_.Name -eq $rname });
+
+        # Get all NatGateways in subscription wildcard for resource group
+        $listNatGateway = Get-AzNatGateway -ResourceGroupName "*";
+        Assert-NotNull ($listNatGateway | Where-Object { $_.ResourceGroupName -eq $rgname -and $_.Name -eq $rname });
+
+        # Get all NatGateways in subscription wildcard for name
+        $listNatGateway = Get-AzNatGateway -Name "*";
+        Assert-NotNull ($listNatGateway | Where-Object { $_.ResourceGroupName -eq $rgname -and $_.Name -eq $rname });
+
+        # Get all NatGateways in subscription wildcard for both resource group and name
+        $listNatGateway = Get-AzNatGateway -ResourceGroupName "*" -Name "*";
+        Assert-NotNull ($listNatGateway | Where-Object { $_.ResourceGroupName -eq $rgname -and $_.Name -eq $rname });
+
+		# Create Subnet
+        $subnet = New-AzVirtualNetworkSubnetConfig -Name $subnetName -AddressPrefix 10.0.1.0/24 -NatGateway $vNatGateway
+        New-AzvirtualNetwork -Name $vnetName -ResourceGroupName $rgname -Location $location -AddressPrefix 10.0.0.0/16 -Subnet $subnet
+        $vnet = Get-AzvirtualNetwork -Name $vnetName -ResourceGroupName $rgname
+
+		# Get Subnet
+		$subnet2 = Get-AzvirtualNetwork -Name $vnetName -ResourceGroupName $rgname | Get-AzVirtualNetworkSubnetConfig -Name $subnetName;
+
+        # Remove NatGateway
+        $job = Remove-AzNatGateway -ResourceGroupName $rgname -Name $rname -PassThru -Force -AsJob;
```

Comment:
Need to remove corresponding subnet or full vnet first as existing reference in subnet blocks deletion of NatGateway